pkg/lightning/backend/local/duplicate.go

```diff
@@ -0,0 +1,637 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package local
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"sort"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	split "github.com/pingcap/br/pkg/restore"
+
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/distsql"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/ranger"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/keepalive"
+
+	backendkv "github.com/pingcap/br/pkg/lightning/backend/kv"
+	"github.com/pingcap/br/pkg/lightning/common"
+	"github.com/pingcap/br/pkg/lightning/log"
+	"github.com/pingcap/br/pkg/logutil"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/pingcap/errors"
+	sst "github.com/pingcap/kvproto/pkg/import_sstpb"
+	kvrpc "github.com/pingcap/kvproto/pkg/kvrpcpb"
+	tikv "github.com/pingcap/kvproto/pkg/tikvpb"
+)
+
+const (
+	maxWriteBatchCount = 128
+)
+
+type DuplicateRequest struct {
+	tableID   int64
+	indexID   int64 // 0 represent it is a table request
```

Comment:
according to TiDB's `executor/builder.go` it uses `-1`, not `0`, for the "common handles" (grep for `indexID == -1`). 

so i'm not sure if `indexID == 0` will ever be taken up for a legit secondary index.

Reply:
It seems that this field is never used... I have removed it.